cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SubqueryExpressionsTest.scala

```diff
@@ -0,0 +1,250 @@
+            #Variables introduced by the `MATCH` clause and the outside scope can be used in this scope.""".stripMargin('#'))
+        query(
+          """MATCH (person:Person)
+            #WHERE EXISTS {
+            #  MATCH (person)-[:HAS_DOG]->(dog:Dog)
+            #  WHERE person.name = dog.name
+            #}
+            #RETURN person.name AS name""".stripMargin('#'),
+          ResultAssertions(r => {
+            r.toList should equal(List(Map("name" -> "Andy")))
+          })) {
+          resultTable()
+        }
+      }
+      section("Nesting `EXISTS` subqueries", "existential-subquery-nesting") {
+        p(
+          """`EXISTS` subqueries can be nested like the following example shows.
+            #The nesting also affects the scopes.
+            #That means that it is possible to access all variables from inside the subquery which are either from the outside scope or defined in the very same subquery.""".stripMargin('#'))
+        query(
+          """MATCH (person:Person)
+            #WHERE EXISTS {
+            #  MATCH (person)-[:HAS_DOG]->(dog:Dog)
+            #  WHERE EXISTS {
+            #    MATCH (dog)-[:HAS_TOY]->(toy:Toy)
+            #    WHERE toy.name = 'Banana'
+            #  }
+            #}
+            #RETURN person.name AS name""".stripMargin('#'),
+          ResultAssertions(r => {
+            r.toList should equal(List(Map("name" -> "Peter")))
+          })) {
+          resultTable()
+        }
+      }
+      section("`EXISTS` subquery outside of a `WHERE` clause", "existential-subquery-outside-where") {
+        p(
+          """`EXISTS` subquery expressions can appear anywhere that an expression is valid.
+            |Here the result if the subquery can find the given pattern is returned.""".stripMargin)
+        query(
+          """MATCH (person:Person)
+            #RETURN person.name AS name, EXISTS {
+            #  MATCH (person)-[:HAS_DOG]->(:Dog)
+            #} AS hasDog""".stripMargin('#'),
+          ResultAssertions(r => {
+            r.toList should equal(List(
+              Map("name" -> "Andy", "hasDog" -> true),
+              Map("name" -> "Timothy", "hasDog" -> false),
+              Map("name" -> "Peter", "hasDog" -> true)
+            ))
+          })) {
+          resultTable()
+        }
+      }
+    }
+
+    section("`COUNT` subqueries", "count-subqueries") {
+      p("A `COUNT` subquery expression can be used to to count the number of results of the subquery.")
+      functionWithCypherStyleFormatting(
+        """COUNT {
+          #  MATCH [Pattern]
+          #  WHERE [Expression]
+          #}""".stripMargin('#'))
+      p("It is worth noting that the `MATCH` keyword can be omitted in such subqueries and that the `WHERE` clause is optional.")
+
+      section("Simple `COUNT` subquery", "count-subquery-simple-case") {
+        p(
+          """Variables introduced by the outside scope can be used in the `COUNT` subquery without importing them
```

Comment:
This sentence is a bit confusing. Was it supposed to be "Variables introduced by the outside scope can be used in the `COUNT` subquery without importing them, >>>as it is<<< necessary with `CALL` subqueries"?

Reply:
This sentence should convey 2 things:
* Variables introduced by the outside scope can be used in the `COUNT` subquery without importing them.
* This is different in `CALL` subqueries. There it is necessary to import to import variables introduced by the outside scope.

I am happy to change "as necessary with `CALL` subqueries" to "as it is necessary with `CALL` subqueries", if you think that is better.

Reply:
Right, got it. But if we say "as it is necessary with `CALL` subqueries", we can leave a margin of ambiguity that might make the sentence look like you're saying that it's the same thing rather than the opposite. What about something like this?

Variables introduced by the outside scope can be used in the `COUNT` subquery without importing them, unlike the case for `CALL` subqueries, as they require importing.